### DISCUSSION TITLE
swapping drawing order

### DISCUSSION THREAD

**Discussion by POPREQ** [bnpr/Malt#147] [Q&A] [answered]
In Malt, is there a way to intentionally swap the order in which objects are drawn?

I have eyes embedded in a face object.
I want the material for the whites of the eyes to conform to the face material and be lit, but be drawn behind the eye object.
In EEvee I can create this by using Alpha Blend mode and the order of the material slots, but I can't think of a way to do the same thing in Malt.

Please help!
I will send you a sample file from Eevee.

![image](https://user-images.githubusercontent.com/58585326/137034392-736e3f9e-2c17-4f07-811e-6f3d01ac1567.png)
[swap_drawing_order.zip](https://github.com/bnpr/Malt/files/7333370/swap_drawing_order.zip)

**Comment by pragma37** [comment, marked as answer]
Transparency in Malt is handled in a different (more precise) way than EEVEE, so it's not possible to use the same technique.

You can, however, change the pixel depth (gl_FragDepth) to get the same effect.
I want to add a *Depth Offset* option to the *PixelOutput*, but for now you can add this node to your *Shader Library*:

```glsl
vec4 depth_offset(vec4 color, float depth_offset)
{
    #if defined(PIXEL_SHADER) && !defined(SHADOW_PASS)
    {
        vec3 offset_position = POSITION - view_direction() * depth_offset;
        float projected_depth = project_point(PROJECTION * CAMERA, offset_position).z;
        float far = gl_DepthRange.far;
        float near = gl_DepthRange.near;
        gl_FragDepth = (((far-near) * projected_depth) + near + far) / 2.0;
    }
    #endif

    return color;
}
```
![imagen](https://user-images.githubusercontent.com/36610999/137144357-6fde3994-7f27-4362-bb79-e7e82f8f4409.png)

**Reply by POPREQ** [reply]
I don't have to use trasparency, and I can use it to draw multiple objects in any order I want.
It's great. It's great!
